flight/tests/misc_math/unittest.cpp

```diff
@@ -237,3 +237,54 @@ TEST_F(CircularModulusDeg, SweepError) {
     ASSERT_NEAR(error, circular_modulus_deg(error + 1080), eps);
   }
 };
+
+// Test fixture for vector2_clip()
+class Vector2Clip : public MiscMath {
+protected:
+  virtual void SetUp() {
+  }
+
+  virtual void TearDown() {
+  }
+};
+
+TEST_F(Vector2Clip, TestScale) {
+  float eps = 0.000001f;
+
+  // Choose a non-integer limit
+  for (int i=0; i<3; i++) {
```

Comment:
So it's open to debate as to whether this should be iterative or not. I wrote it for compactness, but an argument can be made that when unit testing no forensics should ever be performed in order to understand the test and that verbosity is valued above all things. So we could split this into three tests, ```TEST_F(Vector2Clip, LessThanUnity)```, ```TEST_F(Vector2Clip, Unity)```, and ```TEST_F(Vector2Clip, GreaterThanUnity)``` (and, eventually, ```TEST_F(Vector2Clip, InvalidInputs)```) if that's for the better.